File: coin_toss.py

### 1. `coin_toss.py` (Python)

```python
# coin_toss.py — Python версия

import random
import time
import os
import sys
from colorama import init, Fore, Style

init(autoreset=True)

# 3D-представление монеты (вращающиеся стороны)
COIN_FRAMES = [
    """
   _____
  /     \\
 |  (●)  |
 |  (●)  |
  \\_____/
""",
    """
   _____
  /     \\
 |  ( )  |
 |  ( )  |
  \\_____/
""",
    """
   _____
  /     \\
 |  (●)  |
 |  (●)  |
  \\_____/
""",
    """
   _____
  /     \\
 |  ( )  |
 |  ( )  |
  \\_____/
""",
    """
   _____
  /     \\
 |  (●)  |
 |  (●)  |
  \\_____/
"""
]

class CoinToss3D:
    def __init__(self):
        self.heads = 0
        self.tails = 0
        self.total = 0

    def clear_screen(self):
        """Очищает экран."""
        os.system('cls' if os.name == 'nt' else 'clear')

    def animate_coin(self, result):
        """Анимирует вращение монеты в 3D."""
        frames = COIN_FRAMES * 2  # повтор для эффекта
        for frame in frames:
            self.clear_screen()
            print("🪙 3D Coin Toss (Python)")
            print("Бросок...")
            print(frame)
            time.sleep(0.08)

        # Показываем результат с цветом
        self.clear_screen()
        print("🪙 3D Coin Toss (Python)")
        if result == 'heads':
            print(Fore.GREEN + "\n   _____")
            print("  /     \\")
            print(" |  (●)  |")
            print(" |  (●)  |")
            print("  \\_____/")
            print("\n✅ Результат: ОРЁЛ!" + Style.RESET_ALL)
            self.heads += 1
        else:
            print(Fore.RED + "\n   _____")
            print("  /     \\")
            print(" |  ( )  |")
            print(" |  ( )  |")
            print("  \\_____/")
            print("\n❌ Результат: РЕШКА!" + Style.RESET_ALL)
            self.tails += 1
        self.total += 1

        print(f"\n📊 Статистика: Орёл: {self.heads}, Решка: {self.tails}")

    def flip(self):
        """Основной метод броска."""
        result = random.choice(['heads', 'tails'])
        self.animate_coin(result)

    def run(self):
        """Запуск интерактивного режима."""
        print("🪙 3D Coin Toss (Python)")
        print("Нажмите Enter, чтобы бросить монету (q для выхода)")

        while True:
            cmd = input("\n> ").strip().lower()
            if cmd == 'q':
                print("До свидания!")
                break
            self.flip()

def main():
    game = CoinToss3D()
    game.run()

if __name__ == "__main__":
    main()

File: coin_toss.rb
# coin_toss.rb — Ruby версия

COIN_FRAMES = [
  "
   _____
  /     \\
 |  (●)  |
 |  (●)  |
  \\_____/",
  "
   _____
  /     \\
 |  ( )  |
 |  ( )  |
  \\_____/"
]

class CoinToss3D
  def initialize
    @heads = 0
    @tails = 0
    @total = 0
  end

  def clear_screen
    system('clear') || system('cls')
  end

  def animate_coin(result)
    6.times do |i|
      clear_screen
      puts "🪙 3D Coin Toss (Ruby)"
      puts "Бросок..."
      puts COIN_FRAMES[i % 2]
      sleep 0.08
    end

    clear_screen
    puts "🪙 3D Coin Toss (Ruby)"
    if result == 'heads'
      puts "\e[32m"
      puts "   _____"
      puts "  /     \\"
      puts " |  (●)  |"
      puts " |  (●)  |"
      puts "  \\_____/"
      puts "\n✅ Результат: ОРЁЛ!\e[0m"
      @heads += 1
    else
      puts "\e[31m"
      puts "   _____"
      puts "  /     \\"
      puts " |  ( )  |"
      puts " |  ( )  |"
      puts "  \\_____/"
      puts "\n❌ Результат: РЕШКА!\e[0m"
      @tails += 1
    end
    @total += 1
    puts "\n📊 Статистика: Орёл: #{@heads}, Решка: #{@tails}"
  end

  def flip
    result = rand(2) == 0 ? 'heads' : 'tails'
    animate_coin(result)
  end

  def run
    puts "🪙 3D Coin Toss (Ruby)"
    puts "Нажмите Enter, чтобы бросить монету (q для выхода)"

    loop do
      print "> "
      input = gets.chomp.strip.downcase
      break if input == 'q'
      flip
    end
    puts "До свидания!"
  end
end

game = CoinToss3D.new
game.run
